### PR TITLE
feat: Add contact store

### PR DESCRIPTION
**Pull request by muhsin-k**
# Pull Request Template

## Description 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have tested in both Android and iOS platform
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/store/contact/contactService.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { apiService } from '@/services/APIService';
+import type { ContactLabelsAPIResponse, ContactLabelsPayload } from './contactTypes';
+
+export class ContactService {
+  static async getContactLabels(payload: ContactLabelsPayload): Promise<ContactLabelsAPIResponse> {
+    const { contactId } = payload;
+    const response = await apiService.get<ContactLabelsAPIResponse>(`contacts/${contactId}/labels`);
+    return response.data;
+  }
```
Comment (pranavrajs) [inline src/store/contact/contactService.ts:9]: We could just return apiService.get instead data as the data handling is done at the actions.

File: src/store/contact/contactSlice.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  initialState,
+  reducers: {
+    clearAllContacts: contactAdapter.removeAll,
+    addContacts: (state, action) => {
+      const { conversations } = action.payload;
+      const contacts = conversations.map((conversation: Conversation) => conversation.meta.sender);
+      contacts.map((contact: Contact) => {
+        contactAdapter.upsertOne(state, contact);
+        return contact.id;
+      });
```
Comment (pranavrajs) [inline src/store/contact/contactSlice.ts:25]: I'm not sure why we need the conversation object on a contactSlice. It should be synthesized in the action before you pass it here.

File: src/store/contact/contactSlice.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    },
+    addContact: (state, action) => {
+      const conversation = action.payload as Conversation;
+      const contact = conversation?.meta?.sender;
+      if (contact) {
```
Comment (pranavrajs) [inline src/store/contact/contactSlice.ts:30]: Same comment as above.

File: src/store/contact/contactSlice.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        contactAdapter.upsertOne(state, contact);
+      }
+    },
+    updateContactsPresence: (state, action) => {
+      const { contacts } = action.payload;
+      if (!contacts) return;
```
Comment (pranavrajs) [inline src/store/contact/contactSlice.ts:36]: When do we get empty contacts in the payload?

File: src/store/contact/contactSlice.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      Object.values(state.entities).forEach(entity => {
+        if (!entity) return;
```
Comment (pranavrajs) [inline src/store/contact/contactSlice.ts:38]: This check seems unnecessary. Is there a case where the Id is present and the value is null or undefined?

File: src/store/contact/contactSlice.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        const contactId = entity.id;
+        if (!contactId) return;
```
Comment (pranavrajs) [inline src/store/contact/contactSlice.ts:40]: Same here. Is there a case where the object is present without an Id.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Please take a look at the individual comments.

**Comment by muhsin-k** [comment]
@pranavrajs Addressed all the comments.

**Review by pranavrajs — APPROVED** [review]

**Pull request merged**